### PR TITLE
Regenerate travis

### PR DESCRIPTION
**Pull request by phadej**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -75,15 +82,17 @@ install:
...
+  - NOINSTALLEDCONSTRAINTS=${NOINSTALLEDCONSTRAINTS-false}
   - GHCHEAD=${GHCHEAD-false}
   - travis_retry cabal update -v
   - "sed -i.bak 's/^jobs:/-- jobs:/' ${HOME}/.cabal/config"
   - rm -fv cabal.project cabal.project.local
   # Overlay Hackage Package Index for GHC HEAD: https://github.com/hvr/head.hackage
   - |
     if $GHCHEAD; then
-      sed -i.bak 's/-- allow-newer:.*/allow-newer: *:base, *:template-haskell, *:ghc, *:Cabal, *:stm/' ${HOME}/.cabal/config
+      sed -i 's/-- allow-newer: .*/allow-newer: *:base/' ${HOME}/.cabal/config
```
Comment (Bodigrim) [inline .travis.yml]: Seems we still need `allow-newer: *:stm` as well.

Comment (phadej) [reply .travis.yml]: Maybe we can then wait until `async` has new `stm` compatible release.

**Comment by Bodigrim** [comment]
I re-run Travis builds - it is all green now.

**Comment by phadej** [comment]
I see that `async` were uploaded on Thu Sep 6.

**Pull request merged**